File: SwapElements/swap.rb
# swap elements soln in rb for codeeval by steven a dunn

File.open(ARGV[0], 'r').each_line do |line|
    line = line.chomp.split(" : ")
    numbers = line[0].split(' ')
    swaps = line[1].split(", ")

    swaps.each do |swap|
        swap = swap.split('-')
        tmp = numbers[swap[0].to_i]
        numbers[swap[0].to_i] = numbers[swap[1].to_i]
        numbers[swap[1].to_i] = tmp
    end
    puts numbers.join(" ")
end

File: SwapElements/py3/swap.py
# Swap Elements solution in Python 3 by Steven A. Dunn

import sys

for line in open(sys.argv[1], 'r'):
  line = line.rstrip('\n').split(" : ")
  numbers = line[0].split(' ')
  swaps = line[1].split(", ")

  for swap in swaps:
    swap = swap.split('-')
    tmp = numbers[int(swap[0])]
    numbers[int(swap[0])] = numbers[int(swap[1])]
    numbers[int(swap[1])] = tmp

  for i in numbers:
    sys.stdout.write(i + " ")
  print()
